packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkConfigurationExt.kt

```diff
@@ -94,12 +97,15 @@ internal fun ReadableMap.asProxyConfig(): Pair<Proxy, ProxyAuthenticator?>? {
 }
 
 internal fun ReadableArray.asFirstPartyHosts(): Map<String, Set<TracingHeaderType>> {
-    val firstPartyHosts = this.toArrayList() as List<ReadableMap>
-    return firstPartyHosts.mapNotNull<ReadableMap, JSONFirstPartyHost> {
-        val match = it.getString("match")
-        val propagatorTypes = it.getArray("propagatorTypes")
+    val firstPartyHosts = this.toHashMapArrayList()
```

Comment:
I think this can be written this way.

```suggestion
    val firstPartyHosts = this.toList().filterIsInstance<ReadableMap>().toMap()
```

Reply:
I would keep `.toMapList()` implementation, previously called `.toHashMapArrayList()` so that we can log an error whenever object of unexpected types are found in the list.